packages/dds/tree/docs/data-model/README.md

```diff
@@ -0,0 +1,166 @@
+
+- The internal in-memory representation of tree data
+- The serialized tree data sent across the network
+- The format used to persist tree data in storage
+
+While each of the above involve specific encodings of the SharedTree data model, the SharedTree data model itself is not concerned with the byte-level representation (see #10034).
+
+## Requirements
+
+### JSON Interoperability
+
+We take it as a given that the underlying data model for the SharedTree is tree-structured data.
+We also agree that JSON is the modern lingua franca of the web and services.
+We therefore begin with the following requirements:
+
+1. JSON must can efficiently and losslessly be encoded by the underlying SharedTree data model, with a small number of possible caveats (see below).
+3. Deviations from JSON in the underlying SharedTree data model must be well justified.
+
+Consequently, we require that the underlying SharedTree data model can express the following in a natural way:
+
+- Object-like records of named properties
+- Array-like sequences of consecutive items (non-sparse / index agnostic)
+- null, true/false, finite numbers (f64), and strings
+
+#### Potential JSON Caveats
+We currently know of two potential caveats regarding JSON interoperability.
+
+##### Arbitrary Precision Numbers
+As defined, the JSON specification allows numbers to be of arbitrary precision.
+Some systems honor this by parsing JSON into a representation that uses an alternative numeric encoding than IEEE 64b floating point format.
+However, JavaScript's built-in `JSON.parse()` does not.
+Therefore, we do not currently see a need to preserve arbitrary precision numbers in M1.
+
+##### Reserved Key Space
+While we require an injective function from the JSON domain to the SharedTree data model, the SharedTree data model is a superset of JSON.
+Therefore, projecting arbitrary SharedTree data to JSON requires tradeoffs due to the fact that the SharedTree data model is a superset of JSON.
+One potential tradeoff is the use of reserved key space to preserve node ids, types, and augmentations.
+
+### Durable References
+
+The data model must support creating inexpensive durable references to nodes within the tree.
+This includes scenarios like creating "share link" URLs or building graph-like relationships within the tree.
+
+### Schema
+
+The data model must encode sufficient information that the system can efficiently layer a schema-on-write policy on top of the underlying data model (see #9282).
+
+### Augmentation
+
+The data model must allow a subset of collaborators to unobtrusively attach extra-schema data to the tree in a way that is ignored (but preserved) by clients that are unaware of these augmentations.
+
+## Model
+
+### Node
+
+In the SharedTree data model, each addressable piece of data is represented as a tree ***node***.
+There is a single node at the root of the SharedTree that can not be removed or replaced.
```

Comment:
Does this node represent actual content? For example would this node have an application-specific schema associated with it (e.g., "WhiteboardDocument")? If so then I've been assuming this node can be removed/replaced. This may be needed for format migration for example.

If this node does not represent actual content then that needs calling out. It also raises questions about whether this node can be assigned a value, and about what trait it stores the true document contents under. It may also just be simpler not to call out the existence of this node unless it affets the model (does it?).

Reply:
The "static root node" is a nice property for editing, since you have a starting point to insert things under when the tree is empty (where empty means "has only the static root node"). That removes the need for a special "insert root node API"; you only need "insert node under parent" API. For that scheme to work for all trees, that static root node must have a static identity across all trees. Therefore you cannot delete or replace it, in practice (or maybe you can, but then potentially lose your entire tree forever lol). But even in that scheme the question remains - should you be allowed to set its value? It feels useful for lots of things, but also kind of like a "cute hack" and you might be better off just being more explicit and putting the value on some node underneath it. 

I am also pondering your last question. Does the data model care about any of this? The reading and editing API cares. But I think to the model the root node in that scheme is just like any other node.